staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructuredscheme/scheme.go

```diff
@@ -89,6 +89,14 @@ func (t unstructuredObjectTyper) ObjectKinds(obj runtime.Object) ([]schema.Group
 	if _, ok := obj.(runtime.Unstructured); !ok {
 		return nil, false, fmt.Errorf("cannot type %T", obj)
 	}
+	gvk := obj.GetObjectKind().GroupVersionKind()
```

Comment:
what triggered this? do we need to guard against internal version as well?

Reply:
> what triggered this? do we need to guard against internal version as well?

If you can't determine the gvk, you should fail.  Found it when we started using an actual unstructured typer for unstructured requests instead of a hybrid delegating typer.